examples/hello/src/Main.gren

```diff
@@ -0,0 +1,112 @@
+module Main exposing (main)
+
+import Node
+import Bytes exposing (Bytes)
+import Bytes.Encode as BE
+import Dict
+import Stream exposing (Stream)
+import Node.Program as Program exposing (Program)
+import HttpServer as Http
+import Task
+
+
+main : Program Model Msg
+main =
+    Program.define
+        { init = init
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+type alias Model =
+    { stdout : Stream
+    , stderr : Stream
+    , server : Maybe Http.Server
+    }
+
+
+type Msg
+    = CreateServerResult (Result Http.CreateServerError Http.Server)
+    | GotRequest Http.Request
+    | SentChunk Int Http.Request
+
+
+init : Program.AppInitTask { model : Model, command : Cmd Msg }
+init =
+    Program.await Node.initialize <| \nodeConfig ->
+        Program.await Http.initialize <| \serverPermission ->
+            Program.startProgram
+                { model =
+                    { stdout = nodeConfig.stdout
+                    , stderr = nodeConfig.stderr
+                    , server = Nothing
+                    }
+                , command =
+                    Task.attempt CreateServerResult <|
+                        Http.createServer serverPermission
+                            { host = "localhost"
+                            , serverPort = 3000
+                            }
+                }
+
+
+update : Msg -> Model -> { model : Model, command : Cmd Msg }
+update msg model =
+    case msg of
+        CreateServerResult result ->
+            case result of
+                Ok server ->
+                    { model = { model | server = Just server }
+                    , command = Stream.sendLine model.stdout <|
+                        "Server started"
+                    }
+                Err _ ->
+                    { model = model
+                    , command = Stream.sendLine model.stderr
+                        "Server failed to start"
+                    }
+        
+        -- TODO: this needs to have the response object attached
+        -- so we can call .write() etc. on it in the kernel.
+        GotRequest req ->
+            { model = model
+            , command = Cmd.batch
+                [ Stream.sendLine model.stdout <| Debug.toString req.url
+                , Http.respondWith req
```

Comment:
Having trouble picturing how this would behave as a task instead of a command. Do we want some kind of `ResponseFinished` message?

Reply:
Response finished message seems fine and maybe useful